migration_guides/v0.10.2_to_v1.0.0_nearcore.md

```diff
@@ -0,0 +1,375 @@
+```
+
+But the real issue here is that `borsh` doesn't have a generic implementation
+of `BorshDeserialize` for `&'a T`, where `T: borsh::de::BorshDeserialize` (nor did it have it in 0.10.2 version),
+and that the derived `BorshDeserialize` wasn't used (and it couldn't be for such a field's type).
+
+So the right change is to remove `BorshDeserialize` derive from the struct:
+
+```diff
+diff --git a/core/primitives/src/signable_message.rs b/core/primitives/src/signable_message.rs
+index efdd489ac..db97eb1fd 100644
+--- a/core/primitives/src/signable_message.rs
++++ b/core/primitives/src/signable_message.rs
+@@ -57,3 +57,3 @@ pub struct MessageDiscriminant {
+ /// the context in which the message is received.
+-#[derive(BorshSerialize, BorshDeserialize)]
++#[derive(BorshSerialize)]
+ pub struct SignableMessage<'a, T> {
+```
+
+## 6. similar errors
+
+We encounter an error in `near-store` package, similar to above ones.
+It was fixed in a similar way.
+
+### 7. next we encounter an error in `near-network` package:
+
+```rust
+ 1  error: You have to specify `#[borsh(use_discriminant=true)]` or `#[borsh(use_discriminant=false)]` for all enums with explicit discriminant                                                                                                                            ▐
+   --> chain/network/src/types.rs:56:10                                                                                                                                                                                                                                    ▐
+    |                                                                                                                                                                                                                                                                      ▐
+ 56 | pub enum ReasonForBan {                                                                                                                                                                                                                                              ▐
+    |          ^^^^^^^^^^^^                                                                                                                                                                                                                                                ▐
+```
+
+```rust
+#[derive(borsh::BorshSerialize, borsh::BorshDeserialize, Debug, Clone, PartialEq, Eq, Copy)]
+pub enum ReasonForBan {
+    None = 0,
+    BadBlock = 1,
+    BadBlockHeader = 2,
+    HeightFraud = 3,
+    BadHandshake = 4,
+    BadBlockApproval = 5,
+    Abusive = 6,
+    InvalidSignature = 7,
+    InvalidPeerId = 8,
+    InvalidHash = 9,
+    InvalidEdge = 10,
+    InvalidDistanceVector = 11,
+    Blacklisted = 14,
+}
+```
+
+We fix it with `#[borsh(use_discriminant=false)]` as it will result in `12u8` value for `Blacklisted` 
+variant tag, used in serialization, same as before:
```

Comment:
```suggestion
We fix it with `#[borsh(use_discriminant=false)]` to preserve the behaviour of borsh before 1.0 release which serialized `ReasonForBan::Blacklisted` as 12 instead of 14 (borsh 0.10 and older [ignored explicit discriminant values in enum definitions](https://github.com/near/borsh-rs/issues/137)):
```